### DISCUSSION TITLE
Is there a reference with mapping target <-> cpu? Or a way to know which cpu, CC65 assumes for any target?

### DISCUSSION THREAD

**Discussion by Fabrizio-Caruso** [cc65/cc65#2218] [Q&A]
Where can I find a reference/list to know for which specific CPU the CC65 C compiler compiles for a given target?
Or a way to know which cpu, CC65 assumes it has to be used for any target?

In particular for #2210 I would like to see if I am getting the same error on other targets with the exact same CPU (for CC65) if any.
It may even be possible the CC65 is assigning the wrong CPU to Supervision.

**Comment by mrdudz** [comment]
you'll have too look into the sourcecode, i guess

**Reply by Fabrizio-Caruso** [reply]
@mrdudz ok, do you by chance know which files?
It could be useful to have such mapping or at least a list of the supported CPUs.

**Comment by mrdudz** [comment]
https://github.com/cc65/cc65/blob/12193790e5f8238ed9dad0fd007903bf77fca4aa/src/common/target.c#L187

**Comment by Fabrizio-Caruso** [comment]
@mrdudz thanks a lot.
I will double check the Lynx target, then, which, according to the target.c file is the only one that shares the CPU with the Supervision.
Maybe the optimer fails for some other reason unlrelated to the specific CPU on the Supervision.

**Comment by Fabrizio-Caruso** [comment]
@mrdudz  maybe the problem is in target.c, which is assuming the Watara Supervision has a 65SC02 instead of a 65C02 and I fear they may have subtle differences on how some bit operations are performed.

**Comment by Fabrizio-Caruso** [comment]
@mrdudz documentation on line is contradictory on this subject...

**Comment by mrdudz** [comment]
no idea really

**Comment by polluks** [comment]
You may try something like this
grep -r .setcpu * | grep -v 6502


2023-10-07 20:49 GMT+02:00, Bob Andrews ***@***.***>:

**Comment by polluks** [comment]
@Fabrizio-Caruso which documentation?
